site/js/helpers.js

```diff
@@ -135,59 +411,69 @@ function Evaluate (offset, reset) {
-                document.forms.check.elements['cc'].value    = nominatim.address.country_code;
-                document.forms.check.elements['state'].value = nominatim.address.state;
-                Evaluate();
-            }
-        );
+        try {
+            nominatim = await reverseGeocodeLocation(
+                lat,
+                lon,
+                mapCountryToLanguage(i18next.language)
+            );
+            document.forms.check.elements['cc'].value = nominatim.address.country_code;
+            document.forms.check.elements['state'].value = nominatim.address.state;
+            Evaluate();
+        } catch (error) {
+            /* Set fallback Nominatim answer to allow using the evaluation tool even without Nominatim. */
+            console.error('Reverse geocoding failed:', error);
+            alert('Reverse geocoding of the coordinates using Nominatim was not successful. The evaluation of features of the opening_hours specification which depend this information will be unreliable. Otherwise, this tool will work as expected using a fallback answer. You might want to check your browser settings to fix this.');
+            nominatim = {'place_id':'44651229','licence':'Data \u00a9 OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright','osm_type':'way','osm_id':'36248375','lat':'49.5400039','lon':'9.7937133','display_name':'K 2847, Lauda-K\u00f6nigshofen, Main-Tauber-Kreis, Regierungsbezirk Stuttgart, Baden-W\u00fcrttemberg, Germany, European Union','address':{'road':'K 2847','city':'Lauda-K\u00f6nigshofen','county':'Main-Tauber-Kreis','state_district':'Regierungsbezirk Stuttgart','state':'Baden-W\u00fcrttemberg','country':'Germany','country_code':'de','continent':'European Union'}};
+            document.forms.check.elements['cc'].value = nominatim.address.country_code;
+            document.forms.check.elements['state'].value = nominatim.address.state;
+            Evaluate();
+        }
+        return;
     }
 
     date = reset
         ? new Date()
         : new Date(
-            document.forms.check.elements['yyyy'].value,
-            document.forms.check.elements['mm'].selectedIndex,
-            document.forms.check.elements['dd'].value,
-            document.forms.check.elements['HH'].value,
-            parseInt(document.forms.check.elements['MM'].value),
+            window.currentDateTime.year,
+            window.currentDateTime.month,
+            window.currentDateTime.day,
+            window.currentDateTime.hour,
+            window.currentDateTime.minute,
             offset
         );
 
-    function u2 (v) { return v>=0 && v<10 ? '0'+v : v; }
+    // Update global state
+    window.currentDateTime = {
```

Comment:
A central

```js
    export let currentDateTime
```
and here

```suggestion
    currentDateTime = {
```

??

Reply:
Good!

Done, along with a few similar cases.